arangod/StorageEngine/StorageEngine.h

```diff
@@ -119,7 +119,7 @@ class StorageEngine : public application_features::ApplicationFeature {
   // --------------------
 
   // return the name of the specific storage engine e.g. rocksdb
-  char const* typeName() const { return _typeName.c_str(); }
+  virtual char const* typeName() const { return _typeName.c_str(); }
```

Comment:
is virtual required instead of storing vale inside _typeName via constructor?

Reply:
unfortunately the actual engine type is selected later and I did not want to have two different Cluster engine features for each actual storage engine

Reply:
does not look like a clean design